src/tests/tracing/runtimeeventsource/NativeRuntimeEventSourceTest.cs

```diff
@@ -31,14 +32,19 @@ static int Main(string[] args)
                     if (OperatingSystem.IsWindows())
                         DoOverlappedIO();
 
-                    // Wait for events.
-                    Thread.Sleep(1000);
-
                     // Generate some GC events.
                     GC.Collect(2, GCCollectionMode.Forced);
 
-                    // Wait for more events.
-                    Thread.Sleep(1000);
+                    Stopwatch sw = Stopwatch.StartNew();
+
+                    while (sw.Elapsed <= TimeSpan.FromMinutes(1))
+                    {
+                        Thread.Sleep(100);
+                        if (listener.SeenProvidersAndEvents.Contains("Microsoft-Windows-DotNETRuntime/EVENTID(65)"))
```

Comment:
It appears that we don't expect to see the ThreadPoolIOPack event on non-windows platforms which would make the test spin for 1 minute waiting. We probably want a different loop exit condition in that case.

Reply:
Ah, good catch. I can change the condition so it breaks if it's Windows and we see the ThreadPack event, or if it's any other platform and we've seen any events.